gax-java/gax/src/main/java/com/google/api/gax/rpc/StubSettings.java

```diff
@@ -584,8 +597,13 @@ public ApiClock getClock() {
       return clock;
     }
 
+    /**
+     * @return the resolved endpoint when the Builder was created. If other parameters are then set
+     *     in the builder, the resolved endpoint is not automatically updated. The resolved endpoint
+     *     will only be recomputed when the StubSettings is built again.
+     */
     public String getEndpoint() {
-      return clientSettingsEndpoint;
+      return endpointContext.resolvedEndpoint();
```

Comment:
A fully compatible solution might be 
```
// For newBuilder() scenario.
if (endpointContext.equals(EndpointContext.getDefaultInstance())) {
  return clientSettingsEndpoint;
}
return endpointContext.resolvedEndpoint();
```

Reply:
Yep, I think you are right. A valid StubSettings converted to a Builder will still be able to have the resolved endpoint.